File: 49-GroupAnagrams/49-GroupAnagrams.py
# Last updated: 22/2/2026, 10:20:26 pm
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dict=defaultdict(list)
        for s in strs:
            sortedS=''.join(sorted(s))
            dict[sortedS].append(s)
        return list(dict.values())

File: 49-GroupAnagrams/49-GroupAnagrams.cpp
// Last updated: 7/3/2026, 11:36:57 pm
1class Solution {
2public:
3    vector<vector<string>> groupAnagrams(vector<string>& strs) {
4        unordered_map<string, vector<string>> mp;
5        for (string s : strs) {
6            string t = s; 
7            sort(t.begin(), t.end());
8            mp[t].push_back(s);
9        }
10        vector<vector<string>> anagrams;
11        for (auto p : mp) { 
12            anagrams.push_back(p.second);
13        }
14        return anagrams;
15    }
16};